llvm/test/tools/sycl-post-link/sycl-esimd/sycl-esimd-split-per-kernel.ll

```diff
@@ -0,0 +1,64 @@
+; RUN: sycl-post-link -split=kernel -S %s -o %t.table
+; RUN: FileCheck %s -input-file=%t.table
+; RUN: FileCheck %s -input-file=%t_0.ll --check-prefixes CHECK-SYCL-IR-0
+; RUN: FileCheck %s -input-file=%t_1.ll --check-prefixes CHECK-SYCL-IR-1
+; RUN: FileCheck %s -input-file=%t_esimd_0.ll --check-prefixes CHECK-ESIMD-IR-0
+; RUN: FileCheck %s -input-file=%t_esimd_1.ll --check-prefixes CHECK-ESIMD-IR-1
+
+; This test checks that after we split SYCL and ESIMD kernels into
+; separate modules, we split those two modules further according to
+; -split option. In this case we have 2 SYCL and 2 ESIMD kernels, which
+; are split into a total of 4 separate modules.
+
+target datalayout = "e-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024"
+target triple = "spir64-unknown-linux-sycldevice"
+
+declare dso_local spir_func i64 @_Z28__spirv_GlobalInvocationId_xv()
+
+define dso_local spir_kernel void @ESIMD_kernel1() #0 !sycl_explicit_simd !3{
```

Comment:
What is behavior on SYCL_EXTERNAL functions? I understand this is not ESIMD-specific, but if there are gaps then it would be good to file an issue on that at least. This affects ESIMD as well.

Reply:
We discussed it and decided to leave it as it is for now.

Reply:
right, sounds good.